3.evaluate_model/scripts/nbconverted/class_PR_curves.py

```diff
@@ -88,27 +83,135 @@
+single_class_models_dir = pathlib.Path("../2.train_model/models/single_class_models")
+
+# use a list to keep track of scores in tidy long format for each model and dataset combination
+compiled_class_PR_curves = []
+
+# define combinations to test over
+model_types = ["final", "shuffled_baseline"]
+feature_types = ["CP", "DP", "CP_and_DP"]
+evaluation_types = ["train", "test"]
+phenotypic_classes = features_dataframe["Mitocheck_Phenotypic_Class"].unique()
+
+# colors for plotted lines
+data_split_colors = {
+    "CP_and_DP (test)": "#C214CB",
+    "CP_and_DP (train)": "#E88EED",
+    "CP (test)": "#EB4B4B",
+    "CP (train)": "#F8B5B5",
+    "DP (test)": "#5158bb",
+    "DP (train)": "#B5B9EA",
+}
+
+# data from PR curves will be stored in tidy data format (eventually pandas dataframe)
+compiled_SCM_PR_data = []
+
+# iterate over each model type
+# this is not included in itertools product because the PR curves figure needs to be generated for each model type with plt.show()
+for model_type in model_types:
+
+    # create a figure that has 6x5 subplots
+    fig, axs = plt.subplots(6, 5)
+    fig.set_size_inches(15, 18)
+
+    # iterate through each combination of feature_types, evaluation_types, phenotypic_classes
+    for feature_type, evaluation_type, phenotypic_class in itertools.product(
+        feature_types, evaluation_types, phenotypic_classes
+    ):
+
+        # load single class model for this combination of model type, feature type, and phenotypic class
+        single_class_model_path = pathlib.Path(
+            f"{single_class_models_dir}/{phenotypic_class}_models/{model_type}__{feature_type}.joblib"
+        )
+        single_class_model = load(single_class_model_path)
+
+        # load single cell data that will be used for evaluation
+        single_cell_data = get_dataset(
+            features_dataframe, data_split_indexes, evaluation_type
+        )
+
+        # index of phenotypic_class gives us an idea of where it should be plotted
+        phenotypic_class_index = phenotypic_classes.tolist().index(phenotypic_class)
+
+        # plot PR curves for single class models of this model type
+        # this function also returns the PR curve data for this model type
+        PR_data = class_PR_curves_SCM(
+            single_cell_data,
+            single_class_model,
+            fig,
+            axs,
+            phenotypic_class_index,
+            data_split_colors,
+            model_type,
+            feature_type,
+            evaluation_type,
+            phenotypic_class,
+        )
+        compiled_SCM_PR_data.append(PR_data)
+
+    # add title to figure with all subplots
+    fig.suptitle(f"PR curves for {model_type} models")
+    plt.show()
+
+
+# ### Save PR curves from each evaluation (multiclass)
+# 
+
+# In[6]:
+
+
+# compile tidy PR data
+# some thresholds are None because last PR value doesnt correspond to cell dataset (these values are always P=1, R=0), remove these rows from PR data
```

Comment:
```suggestion
# some thresholds are None because last PR value doesn't correspond to cell dataset (these values are always P=1, R=0), remove these rows from PR data
```

Reply:
Added to notebook in https://github.com/WayScience/phenotypic_profiling_model/commit/987e1ec5a50e2751ae962ec53d06265164fcb301